src/controllers/messageController/uploadFile.ts

```diff
@@ -0,0 +1,31 @@
+import { Request, Response, NextFunction } from 'express';
+import { PutObjectCommand } from '@aws-sdk/client-s3';
+import { v4 as uuidv4 } from 'uuid';
+import { s3 } from './utils/configureAWS';
+
+export const uploadFile = async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    const { originalname, buffer, mimetype } = req.file || {};
+    // replace spaces with underscore '_' and add unique prefix
+    const fileName = `${uuidv4()}-${originalname?.replace(/ /g, '_')}`;
+    const fileContent = buffer;
+
+    const params = {
```

Comment:
It's mainly a code quality question: Should I separate this function into smaller ones? 
 saveDataInS3 etc.